app/assets/javascripts/controllers/middleware_topology/middleware_topology_controller.js

```diff
@@ -112,8 +150,25 @@ function MiddlewareTopologyCtrl($scope, $http, $interval, $location, topologySer
             default :
                 return { x: -9, y: -9, height: 18, width: 18, r: 17};
         }
+    }
 
-    };
+    self.getTextPosition = function getTextPosition(d) {
+        switch (d.item.kind) {
+            case "MiddlewareDatasource" :
+                return { x: 0, y: 8};
+            default :
+                return { x: 26, y: 24};
+        }
+    }
+
+    self.getGlyphFeatures = function getGlyphFeatures(d) {
+        switch (d.item.kind) {
+            case "MiddlewareDatasource" :
```

Comment:
this is better to be placed in a css file and not in code. worth to add a new css file for mw topology.
Please see examples including how glyph style is handled [here](https://github.com/ManageIQ/manageiq/blob/master/app/assets/stylesheets/container_topology.css ) and [here](https://github.com/ManageIQ/manageiq/blob/master/app/assets/stylesheets/topology.css)